File: server/src/main/com/broll/gainea/server/core/processing/IGameUpdateReceiver.java
package com.broll.gainea.server.core.processing;

import com.broll.gainea.server.core.battle.BattleContext;
import com.broll.gainea.server.core.battle.BattleResult;
import com.broll.gainea.server.core.battle.RollManipulator;
import com.broll.gainea.server.core.cards.Card;
import com.broll.gainea.server.core.map.Location;
import com.broll.gainea.server.core.objects.MapObject;
import com.broll.gainea.server.core.objects.Unit;
import com.broll.gainea.server.core.player.Player;

import org.apache.commons.lang3.mutable.MutableBoolean;

import java.util.List;

public interface IGameUpdateReceiver {

    void battleIntention(BattleContext context, MutableBoolean cancelFight);

    void battleBegin(BattleContext context, RollManipulator rollManipulator);

    void battleResult(BattleResult result);

    void playedCard(Card card);

    void moved(List<MapObject> units, Location location);

    void spawned(MapObject object, Location location);

    void damaged(Unit unit, int damage);

    void killed(Unit unit, BattleResult throughBattle);

    void earnedStars(Player player, int stars);

    void turnStarted(Player player);

    void roundStarted();

}

File: server/src/main/com/broll/gainea/server/core/processing/IGameUpdateReceiver.kt
package com.broll.gainea.server.core.processing

import com.broll.gainea.server.core.battle.BattleContext
import com.broll.gainea.server.core.battle.BattleResult
import com.broll.gainea.server.core.battle.RollManipulator
import com.broll.gainea.server.core.cards.Card
import com.broll.gainea.server.core.map.Location
import com.broll.gainea.server.core.objects.MapObject
import com.broll.gainea.server.core.objects.Unit
import com.broll.gainea.server.core.player.Player
import org.apache.commons.lang3.mutable.MutableBoolean

interface IGameUpdateReceiver {
    fun battleIntention(context: BattleContext, cancelFight: MutableBoolean)
    fun battleBegin(context: BattleContext, rollManipulator: RollManipulator)
    fun battleResult(result: BattleResult)
    fun playedCard(card: Card)
    fun unitsMoved(objects: List<MapObject>, location: Location)
    fun unitSpawned(obj: MapObject, location: Location)
    fun unitDamaged(unit: Unit, damage: Int)
    fun unitKilled(unit: Unit, throughBattle: BattleResult?)
    fun earnedStars(player: Player, stars: Int)
    fun turnStarted(player: Player)
    fun roundStarted()
}
